### ISSUE TITLE
`Style/MixinGrouping` error for bare `extend`

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#12079]
`Style/MixinGrouping` can fail if RuboCop runs while the statement is incomplete. This is common if a developer is using tooling such as [Ruby LSP](https://github.com/Shopify/ruby-lsp). Ruby LSP will execute RuboCop as long as there are no syntax errors, such as the case here:

```
class Foo
  extend
end
```

## Expected behavior

It should not raise an error.

## Actual behavior

```
An error occurred while Style/MixinGrouping cop was inspecting /.../foo.rb:1:0.
undefined method `source' for nil:NilClass
/Users/andy/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-1.52.1/lib/rubocop/cop/style/mixin_grouping.rb:120:in `separate_mixins'
```

## RuboCop version

```
1.52.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.0) [arm64-darwin22]
```
